src/ORM/Table.php

```diff
@@ -2249,20 +2253,34 @@ protected function _saveMany(iterable $entities, $options = []): iterable
                     }
                 });
         } catch (Exception $e) {
-            $cleanup($entities);
+            $cleanupOnFailure($entities);
 
             throw $e;
         }
 
         if ($failed !== null) {
-            $cleanup($entities);
+            $cleanupOnFailure($entities);
 
             throw new PersistenceFailedException($failed, ['saveMany']);
         }
 
+        $cleanupOnSuccess = function (EntityInterface $entity) {
+            $entity->clean();
+            $entity->setNew(false);
+
+            foreach ($entity->toArray() as $value) {
```

Comment:
Why do we need to iterate the properties of each entity?

Reply:
Because any associated records also need to be cleaned up.

Reply:
I have actually failed to take care of hasMany associated records, fixing it soon.

Reply:
Done